pkg/controller/remotemachineset/remotemachineset_controller_test.go

```diff
@@ -345,9 +378,14 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 
 			if err != nil && !test.expectErr {
 				t.Errorf("unexpected error: %v", err)
+				return
 			}
-			if err == nil && test.expectErr {
-				t.Errorf("expected error but got none")
+			if test.expectErr {
```

Comment:
I find it a bit clearer if you move the `if test.expectErr` above the preceding if.

```
if test.expectErr {
  assert.Error(t, err, "expected error")
  return
}
if !assert.NoError(t, err, "unexpected error") {
  return
}